### ISSUE TITLE
Handle the WCS in a consistent fashion

### ISSUE THREAD

**Issue by lucabaldini** [lucabaldini/ixpeobssim#552]
This is a meta-issue to track the progress on a series of problems that arose right after the first data release---specifically

* issue #523
* issue #548
* issue #550
* issue #518
* issue #538

Here is a message that I sent to the SNR group that summarizes what’s going on

ixpeobssim has always written the WCS information to the output event lists, and used the same information in high-level analysis tools. 

When we received the first Cas A level-2 files, the WCS information for the X and Y columns in the EVENTS extension were missing altogether, which prevented some of the tools to process the files. In order to get up and running I tweaked ixpeobssim in a few places to read the RA\_OBJ and DEC\_OBJ header keywords, instead, and create a WCS object on the fly. In the meantime Pat got in touch with Allyn, who confirmed that this was an oversight on the SOC side. 

A day later we received a new version of the Cas A files, with all the proper WCS header keywords. Unfortunately, the TCUNI fileds were set to 'pixels' instead of 'deg', which again prevented ixpeobssim to use the WCS off the shelf. At that point I deferred any other change to the next data release. 

In the meantime, we realize that being able to tweak the WCS fields in the file header was indeed a useful things to do \(e.g., to manually align the images\) \*provided that ixpeobssim uses the modified information downstream\*. Some of us discovered that the same effect could be achieved by tweaking the RA\_OBJ and DEC\_OBJ keywords---but this is just incidental and I'd argue not a feature that should be relied upon. \(In other words: if we want to modify the file header to adjust the alignment, while we wait for a definitive fix, I think we should do it through the WCS keywords, an not through RA\_OBJ and DEC\_OBJ\). 

Moving forward I'd like to propose that, at the cost of a slight increase in complexity, I modify ixpeobssim in such a way it uses the WCS information **\*when available\***, and fall back to using RA\_OBJ and DEC\_OBJ, writing out a clear warning, whenever the WCS is either missing or incorrect. This will continue working when the SOC corrects the header, and will provide the intended behavior in the transient.

‌

So, in summary: we want to write the WCS in simulated files and read and use in both simulated and real data consistently.

Original url: https://bitbucket.org/ixpesw/ixpeobssim/issues/552
Created on 2022-02-05T14:48:09.498455+00:00
